pkg/tfvars/azure/azure.go

```diff
@@ -117,6 +120,28 @@ func TFVars(sources TFVarsSources) ([]byte, error) {
 		masterDiskEncryptionSetID = masterConfig.OSDisk.ManagedDisk.DiskEncryptionSet.ID
 	}
 
+	var secureVMDiskEncryptionSetID string
+	var securityEncryptionType string
+	if masterConfig.OSDisk.ManagedDisk.SecurityProfile.DiskEncryptionSet.ID != "" {
+		secureVMDiskEncryptionSetID = masterConfig.OSDisk.ManagedDisk.SecurityProfile.DiskEncryptionSet.ID
+	}
+	if masterConfig.OSDisk.ManagedDisk.SecurityProfile.SecurityEncryptionType != "" {
+		securityEncryptionType = string(masterConfig.OSDisk.ManagedDisk.SecurityProfile.SecurityEncryptionType)
+	}
```

Comment:
It doesn't seem like you actually need these variables or these checks. You can just directly assign below like:

```go
SecureVirtualMachineDiskEncryptionSetID: masterConfig.OSDisk.ManagedDisk.SecurityProfile.DiskEncryptionSet.ID
```
I believe the end result is the same, but this is simpler.